File: Extensions/ActionExtension/ActionExtension/ViewController.m
//
//  ViewController.m
//  ActionExtension
//
//  Created by Hari Kunwar on 11/20/14.
//  Copyright (c) 2014 Hari Kunwar. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
{
    NSUserDefaults *_userDefault;
    __weak IBOutlet UILabel *_textLabel;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    _userDefault = [[NSUserDefaults alloc] initWithSuiteName:@"group.harikunwarDev"];

    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(actionUpdate:)
                                                 name:NSUserDefaultsDidChangeNotification
                                               object:nil];
    
    [[NSNotificationCenter defaultCenter]addObserver:self
                                            selector:@selector(actionUpdate:)
                                                name:UIApplicationDidBecomeActiveNotification
                                              object:nil];

}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)actionUpdate:(NSNotification *)notification
{
    NSString *url = [_userDefault objectForKey:@"urlKey"];
    _textLabel.text = url;
}

@end

File: MultitaskSample/MultitaskSample/ViewController.swift
//
//  ViewController.swift
//  MultitaskSample
//
//  Created by Hari Kunwar on 12/14/15.
//  Copyright © 2015 Learning. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var titleLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
        
        self.titleLabel.text = "Size width =" + "\(size.width)" + " height = " + "\(size.height)"
    }


}

